File: kaggle_environments/envs/halite_fleets/starter_bots/java/halite/Configuration.java
package halite;

public class Configuration {
    
    public final int agentTimeout;
    public final int startingHalite;
    public final int size;
    public final int spawnCost;
    public final int convertCost;
    public final float regenRate;
    public final int maxRegenCellHalite;
    public final int randomSeed;

    public Configuration(String rawConfiguration) {
        this.agentTimeout = HaliteJson.getIntFromJson(rawConfiguration, "agentTimeout");
        this.startingHalite = HaliteJson.getIntFromJson(rawConfiguration, "startingHalite");
        this.size = HaliteJson.getIntFromJson(rawConfiguration, "size");
        this.spawnCost = HaliteJson.getIntFromJson(rawConfiguration, "spawnCost");
        this.convertCost = HaliteJson.getIntFromJson(rawConfiguration, "convertCost");
        this.regenRate = HaliteJson.getFloatFromJson(rawConfiguration, "regenRate");
        this.maxRegenCellHalite = HaliteJson.getIntFromJson(rawConfiguration, "maxRegenCellHalite");
        this.randomSeed = HaliteJson.getIntFromJson(rawConfiguration, "randomSeed");
    }
}

File: kaggle_environments/envs/halite_fleets/starter_bots/ts/halite/Configuration.ts
export class Configuration {
    
    public readonly agentTimeout: number;
    public readonly startingHalite: number;
    public readonly size: number;
    public readonly spawnCost: number;
    public readonly convertCost: number;
    public readonly regenRate: number;
    public readonly maxRegenCellHalite: number;
    public readonly randomSeed: number;

    public constructor(rawConfiguration: string) {
        const config = JSON.parse(rawConfiguration);
        this.agentTimeout = config.agentTimeout;
        this.startingHalite = config.startingHalite;
        this.size = config.size;
        this.spawnCost = config.spawnCost;
        this.convertCost = config.convertCost;
        this.regenRate = config.regenRate;
        this.maxRegenCellHalite = config.maxRegenCellHalite;
        this.randomSeed = config.randomSeed;
    }
}